index.d.ts

```diff
@@ -12,13 +12,17 @@ import { IncomingMessage, ServerResponse } from 'http';
 import pino from 'pino';
 import { err, req, res, SerializedError, SerializedRequest, SerializedResponse } from 'pino-std-serializers';
 
-declare function PinoHttp<IM = IncomingMessage, SR = ServerResponse, CustomLevels extends string = never>(opts?: Options<IM, SR>, stream?: pino.DestinationStream): HttpLogger<IM, SR, CustomLevels>;
+declare function PinoHttp< IM = IncomingMessage, SR = ServerResponse, Opts = Options<IM, SR>>(opts?: Opts, stream?: pino.DestinationStream): HttpLogger<IM, SR, Opts>;
 
 declare function PinoHttp<IM = IncomingMessage, SR = ServerResponse>(stream?: pino.DestinationStream): HttpLogger<IM, SR>;
 
-export interface HttpLogger<IM = IncomingMessage, SR = ServerResponse, CustomLevels extends string = never> {
-    (req: IM, res: SR, next?: () => void): void;
-    logger: pino.Logger<CustomLevels>;
+export interface HttpLogger<IM = IncomingMessage, SR = ServerResponse, Opts = Options<IM, SR>> {
+  (req: IM, res: SR, next?: () => void): void;
+  logger: pino.Logger<
+    // this makes custom level loggers typesafe, but TS complains
+    // @ts-expect-error
+    keyof Opts["customLevels"] | keyof Opts["logger"]["customLevels"]
```

Comment:
if anyone can make TS happy via some type gymnastics please do, but from a client perspective this works

Reply:
I'm not convinced of using an expects-error here

Reply:
I understand, I'll see if I can work out the type gymnastics

Reply:
fixed!